receiver/awscontainerinsightreceiver/internal/host/nodeCapacity.go

```diff
@@ -0,0 +1,65 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package host
+
+import (
+	"os"
+
+	"github.com/shirou/gopsutil/cpu"
+	"github.com/shirou/gopsutil/mem"
+	"go.uber.org/zap"
+)
+
+const (
+	GoPSUtilProcDirEnv = "HOST_PROC"
+)
+
+type NodeCapacity struct {
+	MemCapacity int64
+	CPUCapacity int64
+	logger      *zap.Logger
+}
+
+func NewNodeCapacity(logger *zap.Logger) (*NodeCapacity, error) {
+	if _, err := os.Lstat("/rootfs/proc"); os.IsNotExist(err) {
+		return nil, err
+	}
+	if err := os.Setenv(GoPSUtilProcDirEnv, "/rootfs/proc"); err != nil {
```

Comment:
What is this for? I wouldn't expect a receiver to be setting any environment variables

Reply:
It's for using a different `/proc` for gopsutil https://github.com/shirou/gopsutil#:~:text=you%20can%20set%20an%20alternative%20location%20to%20%2Fproc%20by%20setting%20the%20host_proc%20environment%20variable